### PR TITLE
Add ability to delete a report in My Alerts

### PR DESCRIPTION
**Pull request by gopar**
**Description of change**
In the My Alerts section, theres a new column for the context menu which allows the user to view the report or delete it.

**How to test**
Pull down changes and run locally.
Go to My Alerts and you should see the new context menu.
Clicking Delete should soft delete in the backend and remove the report from the view

**Issue(s)**
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-83

**Checklist**

- [x] Meets issue criteria
- [ ] Code tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified

### CODE REVIEW

**Comment by kryswisnaskas** [comment]
This looks pretty good at a first glance. I am still reviewing - so far found that the modal can show up underneath the main table. I thought you might want to see this:

image

**Review by dcloud — COMMENTED** [review]
I took a look at this after reviewing the backend PR. That DeleteReportModal is looking pretty good so far. 

I did see the modal appear under the table, which seems like some `z-index` issues. Which reminded me that I saw similar issues when working on TTAHUB-89 – the ContextMenu was being cut off if it was the menu for the last row in a table. I see the ContextMenu being cut off here (again only the last row is affected). I _think_ the SimpleBar that wraps these tables introduces some z-index weirdness.

File: frontend/src/pages/Landing/index.js [hunk 1]
```diff
@@ -144,9 +136,6 @@ function renderReports(reports, history) {
             {status === 'needs_action' ? 'Needs action' : status}
           </Tag>
         </td>
-        <td>
-          <ContextMenu label={contextMenuLabel} menuItems={menuItems} />
```
[the remark this answers is not in the document]
Comment (gopar) [reply frontend/src/pages/Landing/index.js:148]: yeah I was just fixing those. Looks like its in there to keep. I'll make the chanes in my branch

File: frontend/src/pages/NotFound/__tests__/index.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import '@testing-library/jest-dom';
```
[the remark this answers is not in the document]
Comment (gopar) [reply frontend/src/pages/NotFound/__tests__/index.js:1]: You got me :p 
Also it was to make it not appear anymore in the low coverage part of the report. This didn't really bump the coverage

File: frontend/src/pages/Landing/MyAlerts.js [hunk 3]
```diff
@@ -52,43 +57,80 @@ function renderReports(reports) {
...
+      // eslint-disable-next-line no-constant-condition
+      if (false) deleteReport(reportId);
+      fetchReports();
+      closeModal();
+    };
+    const ConnectModal = connectModal(DeleteReportModal);
+
+    const menuItems = [
+      {
+        label: 'Open report',
```
Comment (jasalisbury) [inline frontend/src/pages/Landing/MyAlerts.js:79]: The table below labels this as `View`, will want to update them to match (can probably also drop `report` on both labels)

File: frontend/src/components/__tests__/DeleteReportModal.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      isOpen
+    />);
+    // When the modal is triggered
+    const buttons = await screen.findAllByRole('button');
+
+    // Then we see our options
+    expect(buttons.length).toBe(2);
+  });
+
+  it('exists when escapse key is pressed', async () => {
```
Comment (jasalisbury) [inline frontend/src/components/__tests__/DeleteReportModal.js:44]:
```suggestion
  it('exits when escapse key is pressed', async () => {
```

**Review by jasalisbury — COMMENTED** [review]
Looks good! Do we want to hide the delete option behind a feature flag until the backend is finished?

File: frontend/src/Constants.js [hunk 5]
```diff
@@ -74,6 +74,7 @@ export const managerReportStatuses = [
 
 export const REPORT_STATUSES = {
   DRAFT: 'draft',
+  DELETED: 'deleted',
```
Comment (gopar) [inline frontend/src/Constants.js:77]: Was included in previous backend PR but for some reason it wasn't merged into the codebase? /shrug

File: frontend/src/pages/Landing/MyAlerts.js [hunk 6]
```diff
@@ -52,43 +57,77 @@ function renderReports(reports) {
...
+    const history = useHistory();
     const idKey = `my_alerts_${id}`;
     const idLink = `/activity-reports/${id}`;
+    const contextMenuLabel = `View activity report ${id}`;
     const statusClassName = `smart-hub--table-tag-status smart-hub--status-${status}`;
 
+    const { isOpen, openModal, closeModal } = useModal();
+    const onDelete = (reportId) => {
+      deleteReport(reportId);
+      fetchReports();
```
Comment (gopar) [inline frontend/src/pages/Landing/MyAlerts.js:69]: For some reason refetchting the reports isn't working, you have to manually refresh to see the report no longer there. Looking into it.

**Comment by kryswisnaskas** [comment]
Looks good. Could we match the delete confirmation box closer to:
https://preview.uxpin.com/ad995fd722bd26c0f9a1c8a115f1b41833637308#/pages/137498615/simulate/no-panels ?

File: frontend/src/pages/Landing/MyAlerts.js (hunk 6 above)
Comment (gopar) [reply frontend/src/pages/Landing/MyAlerts.js:69]: Looks like the DB isn't being updated fast enough for when the a new query is being searched for? They data that is returned when re-fetching the db gives a status of DRAFT for the deleted report, but when checking the db manually after, it's status is set to DELETED. Deployed to staging to check if this happens in deployment. Looks like accounts have been reset in staging tho

**Comment by kryswisnaskas** [comment]
The delete works now and updates the list of alerts! 👍 

The only outstanding item is the "different than design" look of the modal. Other than that it should be good to go.
(https://preview.uxpin.com/ad995fd722bd26c0f9a1c8a115f1b41833637308#/pages/137498615/simulate/no-panels)

**Comment by rahearn** [comment]
What's the status of this? Ready for me to take a look at? (If it is, please also bring it up to date with main before I do)

**Review by kryswisnaskas — APPROVED** [review]

**Review by rahearn — CHANGES_REQUESTED** [review]
delete functionality works well, and looks good for the most part. We just need to implement the fix from #293 on the alerts table

Screen Shot 2021-05-10 at 9 48 25 AM

**Review by rahearn — APPROVED** [review]

**Pull request merged**